Makefile `check` target doesn't exist

## Repro

```console
$ make
make: *** No rule to make target 'check', needed by 'all'.  Stop.
```
## Root cause analysis

#980 added `check` as a target for `all`. It pulls upstream changes, including https://github.com/tendermint/tendermint/pull/3949 by @tac0turtle which removed the `check` target [here](https://github.com/tendermint/tendermint/pull/3949/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52L19).

## Suggested fix

Remove `check` as a dependency for `all`, since `check` hasn't existed since 2019.

--- a/Makefile
+++ b/Makefile
@@ -108,7 +108,7 @@ ifeq (linux/riscv64,$(findstring linux/riscv64,$(TARGETPLATFORM)))
 	GOARCH=riscv64
 endif
 
-all: check build test install
+all: build test install
 .PHONY: all
 
 include tests.mk

